pkg/storegateway/blocks.gohtml

```diff
@@ -90,5 +111,82 @@
     {{ end }}
     </tbody>
 </table>
+<script>
+    // Get the dropdown and elements to control
+    var dropdown = document.getElementById("action-type");
+    var elementsToControl = document.querySelectorAll(".item-to-control");
+
+    // Function to toggle visibility for each element
+    function toggleVisibility(isVisible) {
+        elementsToControl.forEach(function (element) {
+            if (isVisible) {
+                element.removeAttribute("hidden");
+            } else {
+                element.setAttribute("hidden", "hidden");
+            }
+        });
+    }
+
+    // Add an event listener to the dropdown
+    dropdown.addEventListener("change", function () {
+        var isVisible = dropdown.value !== "none";
+        toggleVisibility(isVisible);
+    });
+
+    // Set the initial visibility based on the selected option
+    var isVisible = dropdown.value !== "none";
+    toggleVisibility(isVisible);
+
+    // Function to reload the page with the action
+    function reloadWithAction() {
+        // Get the blocks uid from the table and add it to the form
+        var table = document.getElementById('blocksInfo');
+
+        // Create an array to hold the structured table data
+        var blockULIDs = [];
+
+        // Iterate through the table rows
+        var rows = table.getElementsByTagName('tr');
+        for (var i = 1; i < rows.length; i++) { // Start from 1 to skip header row
+            var cells = rows[i].getElementsByTagName('td');
+
+            // Extract uid from the "Block ID" column (assuming it's the second column)
+            var checkboxStatus = cells[0].getElementsByTagName('input')[0].checked;
+            if (!checkboxStatus) {
+                continue;
+            }
+            // Add the uid to the array
+            blockULIDs.push(cells[1].textContent);
+        }
+
+        // Check if the array is empty
+        if (blockULIDs.length === 0) {
+            // Display an alert popup if the array is empty
+            alert("Select at least one block and then perform the action");
+        }
+        var form = document.getElementById('configForm');    
+        form.querySelector('input[name="block_ulids"]').value = JSON.stringify(blockULIDs);
+        form.method = 'POST';
+        form.submit();
+    }
```

Comment:
Can you explain why all of this is needed? Why can't we handle the form _in the normal html way_?

Reply:
I might have overlooked a simpler method. Since we can select checkboxes in the table, the selected ULIDs should be sent with the request for deletion or updates. However, because the table isn't part of the form, I need to create a variable to parse the table and retrieve the selected ULIDs.

Reply:
What is preventing us from making the table part of the form?

Reply:
we may have a big request body if don't preprocess the table data when we select show parents/ show delete.

Reply:
Only the checked boxes are sent in the form (I obviously had to check this).

Even if all of them were sent, with 26 bytes per ulid, plus the checkbox name, we'd have to send hundreds of thousands of block IDs back to be an issue (assuming 10MB is a reasonable POST). I think your browser will crash before the request size becomes an issue, and you'll have to fallback to the CLI anyway.

Reply:
If we only want checkbox + ulid in the request, we will have to customerise the submit anyway. I was just not convinced it is a nice API design, but if you think it is a cleaner solution to move maximum logics from frontend to backend with all information in the request. I can change to that.

Reply:
> If we only want checkbox + ulid in the request, we will have to customerise the submit anyway. 

Sorry I did not understand you here, can you rephrase please?

> If you think it is a cleaner solution to move maximum logics from frontend to backend with all information in the request. I can change to that.

I'd prefer to have the least possible logic in the frontend as we're not testing it.